Module#02/model/src/main/java/com/epam/esm/dao/Dao.java

```diff
@@ -0,0 +1,17 @@
+package com.epam.esm.dao;
+
+import com.epam.esm.exception.DaoException;
+
+import java.util.List;
+
+public interface Dao<T> {
+    List<T> getAll() throws DaoException;
+
+    T getById(Long id) throws DaoException;
+
+    T getByName(String name) throws DaoException;
+
+    boolean save(T entity) throws DaoException;
+
+    boolean delete(Long id) throws DaoException;
```

Comment:
Don't throw checked exception. it is better to throw unchecked exception (RuntimeException)